src/tools/cephfs/cephfs_shell.py

```diff
@@ -35,73 +36,123 @@ def get_data(file_name):
     data = bytes("",encoding="ascii")
     if fd == -1 :
         return data
-    BUF_SIZE = 65536
-    while True:
-      data = data  + cephfs.read(fd, BUF_SIZE, 1024)
-      if not data:
-        break
+    content =  cephfs.read(fd, 0, cephfs.fstat(fd).st_size)
+    data = data  + content
     return data 
 
 def write_file(file_name, data):
     if not isinstance(file_name, bytes):
       file_name = bytes(file_name, encoding="ascii")
     fd = cephfs.open(file_name, 'w', 0o755)   
     data = bytes(data, encoding="ascii")
+    print(data)
     cephfs.write(fd, data, 0)
 
+def copy_from_local(local_path, remote_path):
+    data = ""
+    for i in open(local_path):
```

Comment:
Where do you read from the local file? 'open' just returns the file object, right?
And I think you'd want open in binary mode.

You'd want to read in byte chunks (size 4MB? not sure about the size) from local file and write that
byte chunk into the CephFS file. Finally close the fd of the local file and the CephFS file.

Reply:
Note that you can't read the entire contents of a very large file at once. What if the file size is much bigger than the machine's memory?